File: src/stubs/java/me/casperge/realisticseasons/api/SeasonsAPI.java
package me.casperge.realisticseasons.api;

import me.casperge.realisticseasons.calendar.Date;
import me.casperge.realisticseasons.season.Season;
import org.bukkit.World;

public class SeasonsAPI {
    private static SeasonsAPI api;

    public static SeasonsAPI getInstance() {
        return api;
    }

    public Date getDate(World world) {
        return null;
    }

    public int getHours(World world) {
        return 0;
    }

    public int getMinutes(World world) {
        return 0;
    }

    public Season getSeason(World world) {
        return null;
    }
}

File: src/stubs/kotlin/me/casperge/realisticseasons/api/SeasonsAPI.kt
package me.casperge.realisticseasons.api

data class SeasonDate(
    val month: Int = 1,
    val day: Int = 1,
    val year: Int = 1,
) {
    fun toString(formatted: Boolean): String =
        if (formatted) {
            "Day $day of Month $month, Year $year"
        } else {
            "$day/$month/$year"
        }
}

class SeasonsAPI private constructor() {
    companion object {
        private val instance: SeasonsAPI = SeasonsAPI()

        @JvmStatic
        fun getInstance(): SeasonsAPI = instance
    }

    fun getDate(defaultWorld: Any): SeasonDate = SeasonDate()

    fun getHours(defaultWorld: Any): Int = 0

    fun getMinutes(defaultWorld: Any): Int = 0

    fun getSeason(defaultWorld: Any): String? = null
}
